### PR TITLE
various minor fixes for verification and approval emails, refs #2089

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | #2089 
| License           | MIT
| Changelog updated | no

### CODE REVIEW

File: src/system/UsersModule/Controller/UserAdministrationController.php [hunk 1]
```diff
@@ -201,40 +202,42 @@ public function approveAction(Request $request, UserEntity $user, $force = false
...
 
                 if ($notificationErrors) {
-                    foreach ($notificationErrors as $error) {
-                        $this->addFlash('error', $error);
-                    }
-                } else {
-                    $this->addFlash('status', $this->__f('Done! %sub% has been approved.', ['%sub%' => $user->getUname()]));
+                    $this->addFlash('error', implode('<br />', $notificationErrors));
                 }
+                $this->addFlash('status', $this->__f('Done! %sub% has been approved.', ['%sub%' => $user->getUname()]));
```
Comment (craigh) [inline src/system/UsersModule/Controller/UserAdministrationController.php]: won't this say `approved` every time? even if there were errors?

File: src/system/UsersModule/Helper/MailHelper.php [hunk 2]
```diff
@@ -101,11 +110,14 @@ public function createAndSendRegistrationMail(UserEntity $userEntity, $userNotif
         }
         if ($adminNotification) {
             // mail notify email to inform admin about registration
-            $mailEmail = $this->variableApi->get('ZikulaUsersModule', UsersConstant::MODVAR_REGISTRATION_ADMIN_NOTIFICATION_EMAIL, '');
-            if (!empty($mailEmail)) {
-                $mailSent = $this->sendNotification($mailEmail, 'regadminnotify', $rendererArgs);
+            $notificationEmail = $this->variableApi->get('ZikulaUsersModule', UsersConstant::MODVAR_REGISTRATION_ADMIN_NOTIFICATION_EMAIL, '');
+            if (!empty($notificationEmail)) {
+                $authMapping = $this->authenticationMappingRepository->getByZikulaId($userEntity->getUid());
```
Comment (craigh) [inline src/system/UsersModule/Helper/MailHelper.php]: not sure about this. Every user will not have an entry in this table. for example, if a user registers with the OAuth module, they will not have an entry. The UsersModule will have a uname and email stored for each user regardless of registration/login method.

File: src/system/UsersModule/Controller/UserAdministrationController.php (hunk 1 above)
Comment (craigh) [inline src/system/UsersModule/Controller/UserAdministrationController.php]: of course 👍

File: src/system/UsersModule/Helper/MailHelper.php [hunk 3]
```diff
@@ -113,7 +113,7 @@ public function createAndSendRegistrationMail(UserEntity $userEntity, $userNotif
             $notificationEmail = $this->variableApi->get('ZikulaUsersModule', UsersConstant::MODVAR_REGISTRATION_ADMIN_NOTIFICATION_EMAIL, '');
             if (!empty($notificationEmail)) {
                 $authMapping = $this->authenticationMappingRepository->getByZikulaId($userEntity->getUid());
-                $rendererArgs['isVerified'] = $authMapping->isVerifiedEmail();
+                $rendererArgs['isVerified'] = null === $authMapping || $authMapping->isVerifiedEmail();
```
Comment (Guite) [inline src/system/UsersModule/Helper/MailHelper.php]: This doesn't get the email, but the information about whether the email has been verified. This is not part of the user entity anymore.

Comment (craigh) [inline src/system/UsersModule/Helper/MailHelper.php]: sorry, I'm not reading the code carefully enough. hard for me to multitask. You're right - email verification is part of ZAuth.

Comment (craigh) [inline src/system/UsersModule/Helper/MailHelper.php]: I guess my only remaining concern about this is that I was trying to separate the users and ZAuth as much as possible with the idea that ZAuth could even possibly be disabled or removed. That may still be impossible with or without this PR but this ties them together more, not less.

Comment (craigh) [inline src/system/UsersModule/Helper/MailHelper.php]: merge it then I guess.

**Pull request merged**